File: solutions/036/36.rb
# Найти сумму чисел, что в 10-м и 2-м предтавлении - палиндромы (585 = 1001001001) - 872187

p 1.step(999_999, 2).inject { |sum, num| (num.to_s == num.to_s.reverse && num.to_s(2) == num.to_s(2).reverse) ? sum + num : sum }

File: solutions/036/36.hs
-- 872187

import Numeric (showIntAtBase)
import Data.Char (intToDigit)

isPalindrome :: String -> Bool
isPalindrome str = str == reverse str

toBinary :: Int -> String
toBinary num = showIntAtBase 2 intToDigit num ""

main :: IO ()
main = print $ sum [x | x <- [1, 3..999999], isPalindrome(show x), isPalindrome(toBinary x)]
